src/core/mesh/qgsmeshdataset.cpp

```diff
@@ -608,31 +636,149 @@ QString QgsMeshDatasetGroupTreeItem::defaultName() const
   return mOriginalName;
 }
 
-QgsMeshDatasetGroupTreeItem::StorageType QgsMeshDatasetGroupTreeItem::storageType() const {return mStoreType;}
+QgsMeshDatasetGroup::Type QgsMeshDatasetGroupTreeItem::datasetGroupType() const
+{
+  return mDatasetGroupType;
+}
 
-void QgsMeshDatasetGroupTreeItem::setStorageType( QgsMeshDatasetGroupTreeItem::StorageType storeType ) {mStoreType = storeType;}
+QString QgsMeshDatasetGroupTreeItem::description() const
+{
+  return mDescription;
+}
+
+void QgsMeshDatasetGroupTreeItem::setDatasetGroup( QgsMeshDatasetGroup *datasetGroup )
+{
+  if ( datasetGroup )
+  {
+    mDescription = datasetGroup->description();
+    mDatasetGroupType = datasetGroup->type();
+    const QStringList &datasetGroupNames = datasetGroup->datasetGroupNamesDependentOn();
+    for ( const QString &varName : datasetGroupNames )
+    {
+      QgsMeshDatasetGroupTreeItem *varItem = searchItemBySourceName( varName );
+      varItem->mDatasetGroupDependencies.append( this->datasetGroupIndex() );
+      mDatasetGroupDependentOn.append( varItem->datasetGroupIndex() );
+    }
+  }
+}
+
+void QgsMeshDatasetGroupTreeItem::setPersistentDatasetGroup( const QString &uri )
+{
+  mDatasetGroupType = QgsMeshDatasetGroup::Persistent;
+  mDatasetGroupDependentOn.clear();
+  mDescription = uri;
+}
 
 QDomElement QgsMeshDatasetGroupTreeItem::writeXml( QDomDocument &doc, const QgsReadWriteContext &context )
 {
   Q_UNUSED( context );
 
   QDomElement itemElement = doc.createElement( QStringLiteral( "mesh-dataset-group-tree-item" ) );
   itemElement.setAttribute( QStringLiteral( "display-name" ), mUserName );
-  itemElement.setAttribute( QStringLiteral( "provider-name" ), mProviderName );
+  itemElement.setAttribute( QStringLiteral( "source-name" ), mSourceName );
   itemElement.setAttribute( QStringLiteral( "original-name" ), mOriginalName );
   itemElement.setAttribute( QStringLiteral( "is-vector" ), mIsVector ? true : false );
   itemElement.setAttribute( QStringLiteral( "dataset-index" ), mDatasetGroupIndex );
   itemElement.setAttribute( QStringLiteral( "is-enabled" ), mIsEnabled ? true : false );
+  itemElement.setAttribute( QStringLiteral( "dataset-group-type" ), mDatasetGroupType );
+  itemElement.setAttribute( QStringLiteral( "description" ), mDescription );
+
+  for ( int index : mDatasetGroupDependentOn )
+  {
+    QDomElement dependOnElement = doc.createElement( QStringLiteral( "dependent-on-item" ) );
+    dependOnElement.setAttribute( QStringLiteral( "dataset-index" ), index );
+    itemElement.appendChild( dependOnElement );
+  }
+
+  for ( int index : mDatasetGroupDependencies )
+  {
+    QDomElement dependencieElement = doc.createElement( QStringLiteral( "dependencie-item" ) );
```

Comment:
typo? "dependencie-item" -> "dependencies-item"

Reply:
there is only one item per element, so dependec**y**